src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,550 @@
+
+using Microsoft.VisualStudio.IO;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Telemetry;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Threading.Tasks;
+
+using Task = System.Threading.Tasks.Task;
+
+using InputTuple = System.Tuple<Microsoft.VisualStudio.ProjectSystem.IProjectSubscriptionUpdate, Microsoft.VisualStudio.ProjectSystem.IProjectSubscriptionUpdate>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    [Export(typeof(ITempPEBuildManager))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasicLanguageService)]
+    internal partial class TempPEBuildManager : UnconfiguredProjectHostBridge<IProjectVersionedValue<InputTuple>, TempPEBuildManager.DesignTimeInputsDelta, IProjectVersionedValue<TempPEBuildManager.DesignTimeInputsItem>>, ITempPEBuildManager
+    {
+        private static readonly TimeSpan s_compilationWaitTime = TimeSpan.FromMilliseconds(500);
+
+        protected readonly IUnconfiguredProjectCommonServices UnconfiguredProjectServices;
+        private readonly IActiveWorkspaceProjectContextHost _activeWorkspaceProjectContextHost;
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IFileSystem _fileSystem;
+        private readonly ITelemetryService _telemetryService;
+        private readonly IProjectFaultHandlerService _projectFaultHandlerService;
+
+        private readonly ITempPECompiler _compiler;
+        private readonly Lazy<VSBuildManager> _buildManager;
+
+        [ImportingConstructor]
+        public TempPEBuildManager(IProjectThreadingService threadingService,
+                                  IUnconfiguredProjectCommonServices unconfiguredProjectServices,
+                                  IActiveWorkspaceProjectContextHost activeWorkspaceProjectContextHost,
+                                  IActiveConfiguredProjectSubscriptionService projectSubscriptionService,
+                                  VSLangProj.VSProjectEvents projectEvents,
+                                  ITempPECompiler compiler,
+                                  IFileSystem fileSystem,
+                                  ITelemetryService telemetryService,
+                                  IProjectFaultHandlerService projectFaultHandlerService)
+             : base(threadingService.JoinableTaskContext)
+        {
+            UnconfiguredProjectServices = unconfiguredProjectServices;
+            _activeWorkspaceProjectContextHost = activeWorkspaceProjectContextHost;
+            _projectSubscriptionService = projectSubscriptionService;
+            _buildManager = new Lazy<VSBuildManager>(() => (VSBuildManager)projectEvents.BuildManagerEvents);
```

Comment:
This looks like a implicit dependency not explicitly called out in an API, someone could change the return value of ProjectEvents.BuildManagerEvents and have no idea you have this dependency.  Introduce a new interface for this or implement VSBuildManager directly.